src/AppProvider.react.js

```diff
@@ -29,13 +29,16 @@ AppProvider.propTypes = {
     registry: PropTypes.shape({
         resolve: PropTypes.func,
     }),
+    name: PropTypes.string,
+    enforceNew: PropTypes.bool,
```

Comment:
Not crazy about the name `enforceNew` - `newStore` or `localStore` perhaps? Or could we just use `name` - if you supply a `name` we make a new store, if not we use the global one?

This makes me wonder what happens if the wider app remounts a dash app component. Could we perhaps cache the store based on `name` so the dash app wouldn't need to reload completely when it's remounted? That might have other issues, dunno, but if we think that's a goal to be working toward, we may want to prepare for it by telling people to use a unique `name` for each dash app on the page.

And... then if we do that (I'm getting a few steps ahead of myself here, I really am curious what you think of all the above and am happy to have a discussion about it, but in case you agree I'll keep going) perhaps `name` is a little less descriptive than we might like - maybe change it to `appKey` or `dashAppKey`?